aiida/transports/transport.py

```diff
@@ -28,6 +28,19 @@
 __all__ = ('Transport',)
 
 
+def validate_positive_number(ctx, param, value):  # pylint: disable=unused-argument
+    """Validate that the number passed to this parameter is a positive number.
+
+    :param ctx: the `click.Context`
+    :param param: the parameter
+    :param value: the value passed for the parameter
+    :raises `click.BadParameter`: if the value is not a positive number
+    """
+    if isinstance(value, (int, float)) and value < 0:
```

Comment:
so, why do you need to check here that it is an int/float when the option requires it to be a float?
does the built-in validation of click happen later?
if yes, then this is indeed backwards - we would like the click validation to go first here.

Reply:
No `click` validation seems to go first, but in some code pathways this gets called with a `None` value. I am thinking now that it is probably called *before* the contextual default is populated. Not sure though

Reply:
Ok, in this case how about transforming this to
`if value is not None and value < 0` ?
or at least you can drop the `int` since you should be getting a `float` here

Reply:
Yeah good point, I like the `is not None` best